File: JS/TakingMaximumEnergyFromTheMysticDungeon.test.js
/**
 * @version JAVASCRIPT ECMAScript 6 
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
import TEST from 'tape'
import { maximumEnergy } from './TakingMaximumEnergyFromTheMysticDungeon.js'

TEST('Starting TakingMaximumEnergyFromTheMysticDungeon test...', (t) => {
    //Test 1
    let want = 3
    let got = maximumEnergy( [5,2,-10,-5,1], 3 )
    t.assert( want === got, "Expect: "+want)
    //Test 2
    want = -1
    got = maximumEnergy( [-2,-3,-1], 2 )
    t.assert( want === got, "Expect: "+want)

    t.end()
})

File: JAVA/TakingMaximumEnergyFromTheMysticDungeonTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class TakingMaximumEnergyFromTheMysticDungeonTest {

    @Test
	public void testTakingMaximumEnergyFromTheMysticDungeon() throws Exception {
        //Test 1
        int want = 3;
        int got = TakingMaximumEnergyFromTheMysticDungeon.maximumEnergy( new int[]{5,2,-10,-5,1}, 3 );
        assertEquals(want, got);
        //Test 2
        want = -1;
        got = TakingMaximumEnergyFromTheMysticDungeon.maximumEnergy( new int[]{-2,-3,-1}, 2 );
        assertEquals(want, got);
	}

}
